packages/babel-preset-fbjs/plugins/inline-requires.js

```diff
@@ -9,144 +9,82 @@
 
 /**
  * This transform inlines top-level require(...) aliases with to enable lazy
- * loading of dependencies.
+ * loading of dependencies. It is able to inline both single references and
+ * child property references.
  *
- * Continuing with the example above, this replaces all references to `Foo` in
- * the module to `require('ModuleFoo')`.
+ * For instance:
+ *     var Foo = require('foo');
+ *     f(Foo);
+ *
+ * Will be transformed into:
+ *     f(require('foo'));
+ *
+ * When the assigment expression has a property access, it will be inlined too,
+ * keeping the property. For instance:
+ *     var Bar = require('foo').bar;
+ *     g(Bar);
+ *
+ * Will be transformed into:
+ *     g(require('foo').bar);
  */
-module.exports = function fbjsInlineRequiresTransform(babel) {
-  var t = babel.types;
-
-  function buildRequireCall(name, inlineRequiredDependencyMap) {
-    var call = t.callExpression(
-      t.identifier('require'),
-      [t.stringLiteral(inlineRequiredDependencyMap.get(name))]
-    );
-    call.new = true;
-    return call;
-  }
-
-  function inlineRequire(path, inlineRequiredDependencyMap, identifierToPathsMap) {
-    var node = path.node;
-
-    if (path.isReferenced()) {
-      path.replaceWith(buildRequireCall(node.name, inlineRequiredDependencyMap));
-      var paths = identifierToPathsMap.get(node.name);
-      if (paths) {
-        paths.delete(path);
-      }
-    }
-  }
-
+module.exports = function fbjsInlineRequiresTransform() {
   return {
     visitor: {
-      Program(_, state) {
-        /**
-         * Map of require(...) aliases to module names.
-         *
-         * `Foo` is an alias for `require('ModuleFoo')` in the following example:
-         *   var Foo = require('ModuleFoo');
-         */
-        state.inlineRequiredDependencyMap = new Map();
-
-        /**
-         * Map of variable names that have not yet been inlined.
-         * We track them in case we later remove their require()s,
-         * In which case we have to come back and update them.
-         */
-        state.identifierToPathsMap = new Map();
-      },
+      CallExpression: function(path) {
+        var declaratorPath = requireAlias(path) || requireMemberAlias(path);
+        var declarator = declaratorPath && declaratorPath.node;
 
-      /**
-       * Collect top-level require(...) aliases.
-       */
-      CallExpression: function(path, state) {
-        var node = path.node;
+        if (declarator) {
+          declaratorPath.scope.crawl();
```

Comment:
The only thing that I don’t understand is why you have to crawl here, and not …

Reply:
We need to crawl because another plugin might have changed the scope already; for instance, the `import` / `export` Babel plugin. We need to have accurate variable binding, counting and referencing for the plugin to work.